onenote/endpoint.bal

```diff
@@ -0,0 +1,274 @@
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+
+# Ballerina Microsoft OneNote connector provides the capability to access Microsoft Graph Excel API
+# It provides capability to perform perform CRUD (Create, Read, Update, and Delete) operations on [OneNote]
+# (https://docs.microsoft.com/en-us/graph/api/resources/onenote-api-overview?view=graph-rest-1.0).
+# 
+# + httpClient - HTTP Client
+@display {
+    label: "Microsoft OneNote Client"
+}
+public isolated client class Client {
+    private final http:Client httpClient;
+
+    # Initializes the connector. During initialization you can pass either http:BearerTokenConfig if you have a bearer 
+    # token or http:OAuth2RefreshTokenGrantConfig if you have OAuth tokens.
+    # Create a Microsoft account and obtain tokens following 
+    # [this guide](https://docs.microsoft.com/en-us/azure/active-directory/develop/active-directory-v2-protocols)
+    # 
+    # + config - Configuration required to initialize the client
+    # + return - An error if initialization fails
+    public isolated function init(Configuration config) returns error? {
+        self.httpClient = check new (BASE_URL, {
+            auth: config.authConfig,
+            secureSocket: config?.secureSocketConfig
+        });
+    }
+
+    // Notebooks
+
+    # Creates a notebook.
+    # 
+    # + noteBookName - Display name for notebook
+    # + return - Notebook if successful or else an error
+    @display {label: "Create Notebook"}
+    remote isolated function createNotebook(@display {label: "Notebook Name"} string noteBookName) returns 
+                                            Notebook|error {
+        string path = NOTEBOOKS_PATH;
+        http:Response response = check self.httpClient->post(path, {"displayName": noteBookName});
+        map<json> handledResponse = check handleResponse(response);
+        return check handledResponse.cloneWithType(Notebook);
+    }
+
+    # Lists notebooks.
+    # 
+    # + oDataQuery - Optional. OData Query. Example: "$top=2&$count=true". 
+    # For details, refer https://docs.microsoft.com/en-us/graph/query-parameters#odata-system-query-options
+    # + return - Notebook[] if successful or else an error
+    @display {label: "List Notebooks"}
+    remote isolated function listNotebooks(@display {label: "OData Query"} string? oDataQuery = ()) 
```

Comment:
How do we address pagination here?  Can't we use streams?

Reply:
User's can use OData query for pagination.
Even in MS Excel connector, we are not using streams.

Reply:
MS Excel doesn't expose next page link to do the pagination. AFAIK, pagination is handled in other MS connectors.

Reply:
Yes. In OneNote also having the same issue we faced in MS Excel.
 It doesn't expose next page link to do the pagination.
 
 So i tested with OData query.
 User's can use this and control the pagination.